### PR TITLE
[ENG-2133] Admin App for Registries moderation

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Add functionality for admin app ahead of registries moderation project.

## Changes

- adds ability to add/remove moderators from provider
- adds a few new fields to edit provider such as moderation type and comment visibility

Screenshot_2020-09-29 Registration Provider(1)

Screen Shot 2020-09-30 at 1 04 44 PM

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify all links go to logically where you would think
- Verify all text inputs are validated and behave correctly
- Verify that everything necesarry is editable on the registration provider.
- Verify moderators can be added/removed correctly
- Verify non-functional requirements are met, if it's anything needs bulk updates etc? 

What are the areas of risk?

Permissions are the greatest risk here, but everything should be acceptable for admins to do, so nothing complicated.

Any concerns/considerations/questions that development raised?

Some scope questions about how much on the registration should be editable.

## Documentation

admin stuff is institutional knowledge! passed down from generation to generation!

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2133

### CODE REVIEW

File: admin/registration_providers/forms.py [hunk 1]
```diff
@@ -20,9 +20,18 @@ class RegistrationProviderForm(forms.ModelForm):
 
     class Meta:
         model = RegistrationProvider
-        exclude = ['primary_identifier_name', 'primary_collection', 'type', 'allow_commenting', 'advisory_board',
-                   'example', 'domain', 'domain_redirect_enabled', 'reviews_comments_anonymous',
-                   'reviews_comments_private', 'reviews_workflow', 'collected_type_choices', 'status_choices']
```
Comment (brianjgeiger) [inline admin/registration_providers/forms.py:25]: Drive-by: I think we don't want `reviews_comments_private` for registries. I also think @jwalz is going to want to do something different from `reviews_workflow`, so I'm pinging him to see if we should keep excluding this field.

Comment (jwalz) [reply admin/registration_providers/forms.py:25]: Planning on still using reviews_workflow, for consistency, just overriding the inherited field to limit the choices

**Review by UdayVarkhedkar — APPROVED** [review]
Looks good and works well!

**Review by brianjgeiger — APPROVED** [review]
